src/Symfony/Component/Security/Core/Encryption/AsymmetricEncryptionInterface.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encryption;
+
+use Symfony\Component\Security\Core\Exception\EncryptionException;
+use Symfony\Component\Security\Core\Exception\WrongEncryptionKeyException;
+
+/**
+ * Asymmetric encryption uses a "key pair" ie a public key and a private key. It
+ * is safe to share your public key but the private key should always be kept a
+ * secret.
+ *
+ * When Alice and Bob wants to communicate thay share the public keys with each
+ * other. Alic will encrypt message with keypair [ alice_private, bob_public ].
+ * When Bob receive the message, he will decrypt it with keypair [ bob_private, alice_public ].
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental in 5.2
+ */
+interface AsymmetricEncryptionInterface
+{
+    /**
+     * Generate new a keypair to be used with encryption. Don't loose your private
+     * key and make sure to keep it a secret.
+     *
+     * @return array{public: string, private: string}
```

Comment:
We are not using this format anywhere if I'm not wrong

Reply:
True, and we never define the keys of an array either. The best I can do following our current pattern is to add `array<string,string>`.

I could also use ArrayShape php8 attribute from [PHPStorm](https://packagist.org/packages/jetbrains/phpstorm-attributes) like: `#[ArrayShape([‘ public’ => ‘string’, ‘ private’ => ‘Foo’,])]`. 

Or, define a small object for a `Keypair`. 

I'll go either way.